File: src/main/java/dev/kikugie/techutils/command/IsorenderSelectionCommand.java
package dev.kikugie.techutils.command;

import com.mojang.brigadier.CommandDispatcher;
import com.mojang.brigadier.context.CommandContext;
import fi.dy.masa.litematica.data.DataManager;
import net.fabricmc.fabric.api.client.command.v2.FabricClientCommandSource;
import net.fabricmc.fabric.impl.command.client.ClientCommandInternals;
import net.fabricmc.loader.api.FabricLoader;
import net.minecraft.command.CommandRegistryAccess;
import net.minecraft.text.Text;

import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.literal;

public class IsorenderSelectionCommand {
	public static void register(CommandDispatcher<FabricClientCommandSource> dispatcher, CommandRegistryAccess ignoredAccess) {
		if (FabricLoader.getInstance().isModLoaded("isometric-renders")) {
			dispatcher.register(literal("isorender").then(literal("selection").executes(IsorenderSelectionCommand::renderLitematicaSelection)));
		}
	}

	@SuppressWarnings("UnstableApiUsage")
	private static int renderLitematicaSelection(CommandContext<FabricClientCommandSource> context) {
		var selection = DataManager.getSelectionManager().getCurrentSelection();
		if (selection == null) {
			context.getSource().sendError(Text.of("No Litematica selection!"));
			return 1;
		}
		var box = selection.getSelectedSubRegionBox();
		if (box == null) {
			context.getSource().sendError(Text.of("Invalid Litematica selection!"));
			return 1;
		}

		ClientCommandInternals.executeCommand(String.format("isorender area %d %d %d %d %d %d",
			box.getPos1().getX(),
			box.getPos1().getY(),
			box.getPos1().getZ(),
			box.getPos2().getX(),
			box.getPos2().getY(),
			box.getPos2().getZ()));

		return 1;
	}
}

File: src/client/kotlin/dev/kikugie/techutils/impl/IsorenderSelectionCommand.kt
package dev.kikugie.techutils.impl

import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.context.CommandContext
import dev.kikugie.malilib_extras.util.translate
import dev.kikugie.techutils.util.PlacementUtils
import net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.literal
import net.fabricmc.fabric.api.client.command.v2.ClientCommandRegistrationCallback
import net.fabricmc.fabric.api.client.command.v2.FabricClientCommandSource
import net.minecraft.client.MinecraftClient
import net.minecraft.command.CommandRegistryAccess
import net.minecraft.text.Text

object IsorenderSelectionCommand {
    fun register() {
        ClientCommandRegistrationCallback.EVENT.register(::register)
    }

    fun register(dispatcher: CommandDispatcher<FabricClientCommandSource>, access: CommandRegistryAccess) {
        dispatcher.register(literal("isorender").then(literal("selection").executes(::renderSelection)))
    }

    private fun renderSelection(context: CommandContext<FabricClientCommandSource>): Int {
        val box = PlacementUtils.selection?.selectedSubRegionBox
        if (box == null || box.pos1 == null || box.pos2 == null)
            context.source.sendError(Text.of("techutils.selection_render.no_selection".translate()))
        else MinecraftClient.getInstance().networkHandler
            ?.sendCommand("isorender area ${box.pos1!!.x} ${box.pos1!!.y} ${box.pos1!!.z} ${box.pos2!!.x} ${box.pos2!!.y} ${box.pos2!!.z}")
        return 1
    }
}